### PR TITLE
On tag publish with GitHub Actions (GitHub Release + NPM)

### LINKED ISSUE #320: migrate to GitHub Actions for building and publishing releases
Looks like `travis-ci.org` is being closed down at the end of the year, and projects are being [migrated](https://docs.travis-ci.com/user/migrate/open-source-repository-migration) to `travis-ci.com`.

While a [free tier remains on travis-ci.com](https://travis-ci.com/plans) it seems this may quickly consume the free credits when having to build each on macOS, Windows and Linux for a run.

Let's take this opportunity to get a simple Actions workflow setup:

 - [x] on pull request and push to default branch, build and test on each platform - #323 
 - [ ] on tag, build, package and publish the binaries to GitHub
 - [ ] stretch goal: on tag, publish to NPM

Open questions:

 - ~~will the Electron ARM64 work in #319 continue to work fine on Actions?~~ not currently

### PR DESCRIPTION
**Pull request by dennisameling**
Closes #320 

✔️ on tag, upload all prebuilds to GitHub
✔️ on tag, and after successful build + prebuild upload, publish version on NPM

The only thing you have to do is to go to the repo settings > Secrets > create a secret called "NPM_AUTH_TOKEN" with a token that you got from https://www.npmjs.com/. The GITHUB_TOKEN is created automatically by GitHub Actions.

You can see it in action at https://github.com/dennisameling/node-keytar/actions/runs/373034035.

GH Release example: https://github.com/dennisameling/node-keytar/releases/tag/v7.2.5-beta 
NPM example: https://www.npmjs.com/package/@dennisameling/keytar-temp

When this PR is merged, all you have to do is update the version in `package.json`, then `git tag v7.x.x && git push v7.x.x`, then the pipeline will trigger.

I added `npm-publish` as a separate step that depends on the `build` job, to be sure that all prebuilds have been built and uploaded successfully on **all OSes** prior to publishing to NPM. This only adds a small overhead which I think is reasonable.

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (shiftkey) [inline .github/workflows/ci.yml:118]: I _think_ this could be distilled down to just `libsecret-1-dev` as `npm install` only needs to build from source. But I'll leave this as-is for now and we can refine it further after doing some more testing.

File: .github/workflows/ci.yml
Comment (shiftkey) [inline .github/workflows/ci.yml:131]: I don't currently have access to add additional settings for this repository, so I can't merge this as-is

**Review by shiftkey — CHANGES_REQUESTED** [review]
Thanks for contributing this @dennisameling!

There's at least one blocker on merging and using this, which is why I've had to `Request changes` to ensure we don't land this before it'll properly work.

**Comment by dennisameling** [comment]
@shiftkey I just updated the PR so that it only needs `libsecret-1-dev` in the NPM publish job, as per your suggestion. Can confirm it works.

Release: https://github.com/dennisameling/node-keytar/releases
GH Actions run: https://github.com/dennisameling/node-keytar/actions/runs/430631719

As soon as you have enough permissions to add the NPM secret to the repo, this PR should be good to go 👍🏼 have a great weekend!

File: .github/workflows/ci.yml
Comment (orgads) [reply .github/workflows/ci.yml:124]: Who can set it or grant you access?

**Comment by orgads** [comment]
Need to rebase, as .travis.yml has been changed.

**Comment by dennisameling** [comment]
Rebased 👍🏼

**Review by shiftkey — APPROVED** [review]

**Comment by shiftkey** [comment]
Thanks for the contribution @dennisameling! I plan to do a test release tomorrow to confirm this is all working as expected, before I look at landing some of the newer PRs here...

**Pull request merged**
